### PR TITLE
Fix branch filter pattens for provider branches in ci workflow

### PR DESCRIPTION
**Pull request by potiuk**
I **hope** this time it will be ok:

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,9 +21,15 @@ on:  # yamllint disable-line rule:truthy
   schedule:
     - cron: '28 1,7,13,19 * * *'
   push:
-    branches: ['v[0-9]+-[0-9]+-test', 'providers-*/v[0-9]+-[0-9]+]']
+    branches:
```
Comment (shubhamraj-git) [inline .github/workflows/ci.yml:24]: Should we include quotes for safe YAML parsing

**Review by gopidesupavan — APPROVED** [review]
cool :) easy to look through now..

**Comment by gopidesupavan** [comment]
Tested here, yes hope it works this time :)

image

https://regextester.github.io/

**Comment by potiuk** [comment]
> Tested here, yes hope it works this time :)
> 

The problem is ... it's not regexp. It's not glob either. It's something GitHub speciific https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

For example they only support select ranges lile `[a-z]` ...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (potiuk) [reply .github/workflows/ci.yml:24]: We had discussion about it with @jscheffl recently - I think it does not change safety, and while I favoured quoting, I   see how a bit nicer it is when you can skip them.

@jscheffl -> consider I've learned something new :)

Comment (jscheffl) [reply .github/workflows/ci.yml:24]: YAML is made to be less boilerplate. No quotes needed except if you want to force bool/int to be string. I favor w/o quotes. But might be a matter of taste. No difference in security/safety. All YAML parsers work without. Else it is a bad YAML parser.

--> https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml ?

Comment (shubhamraj-git) [reply .github/workflows/ci.yml:24]: Ohh, thats good.
Thanks for the insights!

Comment (jscheffl) [reply .github/workflows/ci.yml:24]: Mhm, but reading the YAML specs in https://yaml.org/spec/1.2.2/#plain-style I _think_ @shubhamraj-git is right. If the literals `[` and `]` arecontained then it is advised to quote it:

> In addition, inside [flow collections](https://yaml.org/spec/1.2.2/#flow-collection-styles), or when used as [implicit keys](https://yaml.org/spec/1.2.2/#example-single-pair-explicit-entry), plain scalars must not contain the “[”, “]”, “{”, “}” and “,” characters.

Comment (potiuk) [reply .github/workflows/ci.yml:24]: Well. Just to have some interesting discussion on specs:

> Mhm, but reading the YAML specs in https://yaml.org/spec/1.2.2/#plain-style I think @shubhamraj-git is right. If the literals [ and ] arecontained then it is advised to quote it:

>> In addition, inside [flow collections](https://yaml.org/spec/1.2.2/#flow-collection-styles), or when used as [implicit keys](https://yaml.org/spec/1.2.2/#example-single-pair-explicit-entry), plain scalars must not contain the “[”, “]”, “{”, “}” and “,” characters.

* Yeah. "flow collections" was what it was before (and it was quoted) - but I just changed it to `block collections' :)
* Implicit keys - are different things.

So we are safe :)

**Pull request merged**